docs/book/src/simple-external-plugin-tutorial/testdata/sampleexternalplugin/v1/external/types.go

```diff
@@ -0,0 +1,105 @@
+/*
```

Comment:
I think we might need here the @everettraven thoughts.
However, why we are adding this file?
Is not it implemented in kubebuilder itself?
Could we not import kubebuilder as lib?

Reply:
I believe these types _are_ already implemented in Kubebuilder as a library under `pkg/plugin/external`. I'm _assuming_ the reasoning behind needing to copy them over was that @Eileen-Yu was unable to use the changes made in the library without there being a commit to update the go.mod to point to. That being said, I _think_ we should be able to use the go.mod `replace` directive to point to the local version of Kubebuilder with something like:
```
replace (
    sigs.k8s.io/kubebuilder/v3 => {relative path to project root}
)
```

A `replace` directive like that would allow the project to use the changes made directly in the project without having to do stuff like this where we are copying code over.

Reply:
>replace seems a solution for local dev env.

I would say that this is true for this particular use case but there are other uses that are valid for it being pinning versions, using forks, etc. That being said, I don't think it is "wrong" to use the replace directive in the "local dev env" context here specifically because this is a sample that will likely never exist outside of this repository. If it makes it easier to make changes to this sample to have this replace directive then it is worth it IMO.

>If this PR contains the lines of replace, will it be good to be merged?

I haven't done a full review of this PR so I can't say it will be good to be merged but I think it would definitely be closer in my book. I'm not really a fan of doing the copying unless we _have_ to